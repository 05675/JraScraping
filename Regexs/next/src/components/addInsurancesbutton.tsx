import React from 'react';
import { StyledButton } from '@src/styles';
import { StyledPlusSVG } from '@src/styles/svg';

type AddInsurancesButtonProps = {
  label?: string;
  important?: boolean;
  primary?: boolean;
  warning?: boolean;
  smallSize?: boolean;
  fontSize?: string;
  width?: string;
  height?: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onClick?: (event: any) => void;
};

export const AddInsurancesButton: React.FC<AddInsurancesButtonProps> = props => {
  return (
    <>
      <div
        style={{
          position: 'relative',
          width: `${props.width ?? '100%'}`,
          height: `${props.height ?? '48px'}`,
        }}>
        <span className='plus-icon'>
          <StyledPlusSVG color='#525252' />
        </span>
        <StyledButton
          important={props.important}
          primary={props.primary}
          warning={props.warning}
          smallSize={props.smallSize}
          fontSize={props.fontSize}
          width={props.width}
          height={props.height}
          onClick={props.onClick}
          labelOffsetLeft='48px'
          noShadow>
          {props.children}
        </StyledButton>
      </div>

      <style jsx>
        {`
          .plus-icon {
            position: absolute;
            width: 24px;
            height: 24px;
            top: calc(50% - 12px);
            left: 24px;
          }
        `}
      </style>
    </>
  );
};

export default AddInsurancesButton;
