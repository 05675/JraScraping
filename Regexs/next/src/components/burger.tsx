import React from 'react';
import { StyledBurger } from '@src/styles/burger';

/**
 * バーガーボタン用インターフェース
 *
 * @interface BurgerProps
 */
interface BurgerProps {
  open: boolean;
  setOpen: (newValue: boolean) => void;
}

/**
 * バーガーボタンのヘッダーコンポーネント
 *
 * @param {BurgerProps} props {
 *     open
 *     setOpen
 * }
 * @returns {*} Reactコンポーネント
 */
export const BurgerComponent: React.FC<BurgerProps> = props => {
  return (
    <StyledBurger open={props.open} onClick={() => props.setOpen(!props.open)}>
      <div />
      <div />
      <div />
    </StyledBurger>
  );
};
