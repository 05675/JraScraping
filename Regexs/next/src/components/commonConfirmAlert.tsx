import React from 'react';
import { confirmAlert } from 'react-confirm-alert';

interface AlertButton {
  label: string;
  labelColor?: string;
  clickHandler?: Function;
}

interface CommonConfirmAlertProps {
  message: string;
  leftButton: AlertButton;
  rightButton: AlertButton;
}
export const CommonConfirmAlert = ({
  message,
  leftButton,
  rightButton,
}: CommonConfirmAlertProps) => () =>
  confirmAlert({
    customUI: ({ onClose }) => {
      return (
        <div className='react-confirm-alert-body'>
          <p>{message}</p>
          <div className='react-confirm-alert-button-group'>
            <button
              style={{ color: leftButton.labelColor || '#525252' }}
              onClick={() => (leftButton.clickHandler && leftButton.clickHandler()) || onClose()}>
              {leftButton.label}
            </button>
            <button
              style={{ color: rightButton.labelColor || '#525252' }}
              onClick={() => (rightButton.clickHandler && rightButton.clickHandler()) || onClose()}>
              {rightButton.label}
            </button>
          </div>
        </div>
      );
    },
  });
