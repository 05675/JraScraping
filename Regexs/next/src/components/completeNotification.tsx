import React from 'react';

type notificationProps = {
  message: string;
  isNotification: boolean;
  setIsNotification: (newValue: boolean) => void;
};

export const CompleteNotification: React.FC<notificationProps> = (props) => {
  if (props.isNotification) setTimeout(() => props.setIsNotification(false), 4000);
  const handleClick = () => props.setIsNotification(!props.isNotification);
  return (
    <>
      <div className='notif-container' onClick={handleClick} role='button'>
        <p>{props.message}</p>
        <style jsx>
          {`
            .notif-container {
              width: 100%;
              height: 48px;
              display: flex;
              border-radius: 8px;
              align-items: center;
              background: #525252;
              text-align-center;
              transform: ${props.isNotification ? 'translateY(0)' : 'translateY(100%)'};
              opacity: ${props.isNotification ? '1' : '0'};
              pointer-events: ${props.isNotification ? 'auto' : 'none'};
              transition: transform 0.2s ease-in-out, opacity 0.23s ease-in-out;
            }
            .notif-container p {
              font-style: normal;
              font-weight: normal;
              font-size: 12px;
              line-height: 100%;
              letter-spacing: 0.06em;
              color: #ffffff;
              padding-left: 8px;
              width: 100%;
            }
          `}
        </style>
      </div>
    </>
  );
};

export default CompleteNotification;
