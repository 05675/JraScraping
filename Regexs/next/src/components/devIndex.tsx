import * as React from 'react';
import Link from 'next/link';

// テスト用 sample
const pathnameToPageName: { [pathname: string]: string } = {
  // Developers
  '/Developers/createTask': 'タスク作成',
  '/Developers/demoTask': 'タスク一括登録',
  '/Developers/demoControl': 'デモ用管理機能',
  '/Developers/taskManage': '年末調整',

  // 従業員page
  '/signin': 'サインイン',
  '/tasks': 'タスク一覧',
  '/nencho/test_id_0xd1': '2020年分年末調整',
  '/nencho/test_id_0xd1/insurances': '保険料控除',
  '/nencho/test_id_0xd1/lifeInsuranceInputs/new': '生命保険入力',
};

const DevIndex: React.FC = () => {
  return (
    <div>
      <ul className='list-style'>
        <p style={{ textAlign: 'center', color: '#525252' }}>Sample Page（開発用）</p>
        {Object.keys(pathnameToPageName).map(pathname => (
          <div key={pathname}>
            <li className='listItem-style'>
              <Link href={pathname}>
                <a className='Anchor-style'>{pathnameToPageName[pathname]}</a>
              </Link>
            </li>
          </div>
        ))}
      </ul>
      <style jsx>
        {`
          .list-style {
            margin: 0;
            padding: 0;
            position: relative;
          }
          .listItem-style {
            color: black;
            border-left: solid 8px orange;
            background: whitesmoke;
            margin-bottom: 5px;
            line-height: 1.5;
            border-radius: 0 15px 15px 0;
            padding: 0.5em;
            list-style-type: none !important;
          }
          .anchor-style {
            display: block;
            padding: 5px;
          }
        `}
      </style>
    </div>
  );
};

export default DevIndex;
