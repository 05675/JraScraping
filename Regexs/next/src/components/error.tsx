import React from 'react';

interface ErrorProps {
  message: string;
}

export const Error: React.FC<ErrorProps> = props => {
  return (
    <>
      <p className='error'>{props.message}</p>{' '}
      <style jsx>
        {`
          .error {
            font-style: normal;
            font-weight: normal;
            font-size: 12px;
            line-height: 150%;
            letter-spacing: 0.06em;
            color: #fa3939;
            margin: 16px auto 32px auto;
          }
        `}
      </style>
    </>
  );
};
