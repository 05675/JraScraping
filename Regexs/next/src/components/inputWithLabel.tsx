import React from 'react';
import { StyledLabel, StyledErrorText, StyledInput } from '@src/styles';
import { StyledErrorSVG } from '@src/styles/svg';
import { InputWithLabelProps } from '@src/interfaces/lifeInsuranceInputs';

export const InputWithLabel: React.FC<InputWithLabelProps> = props => {
  return (
    <div style={{ marginBottom: '32px' }}>
      <div className='label'>
        <StyledLabel id='titleLabel'>{props.label}</StyledLabel>
      </div>
      <div style={{ position: 'relative' }}>
        <StyledInput id='inputField' {...props} />
        {props.isError && (
          <span className='errorIcon'>
            <StyledErrorSVG id='errorIcon' color='#fa3939' />
          </span>
        )}
      </div>
      {props.isError && (
        <div className='errorText'>
          <StyledErrorText id='errorText'>{props.errorMessage}</StyledErrorText>
        </div>
      )}

      {/* style */}
      <style jsx>{`
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      `}</style>
    </div>
  );
};
