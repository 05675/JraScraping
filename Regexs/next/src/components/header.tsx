import React, { useState, useRef } from 'react';
import { MenuComponent } from '@src/components/menu';
import { BurgerComponent } from '@src/components/burger';
import { useOnClickOutside } from '@src/util/hooks';
import { StyledLogoWhiteSVG, StyledArrowLeftSVG } from '@src/styles/svg';
import Headroom from 'react-headroom';
import Router from 'next/router';
import { CommonConfirmAlert } from '@src/components/commonConfirmAlert';

interface HeaderProps {
  previousPageName: string;
  previousPathname: string;
}

/**
 * 共通ヘッダーコンポーネント
 *
 * @returns {*} Reactコンポーネント
 */
export const Header: React.FC<HeaderProps> = ({ previousPageName, previousPathname }) => {
  const [open, setOpen] = useState(false);
  const node = useRef<HTMLDivElement>(null);
  useOnClickOutside(node, () => setOpen(false));

  const isLifeInpuranceInputPage = previousPageName === '保険料控除';

  const goToPreviousPath = () => Router.push(previousPathname);

  const showGoBackAlertDialog = CommonConfirmAlert({
    message: 'この保険情報を破棄してもよろしいですか？',
    leftButton: {
      label: '破棄',
      labelColor: '#FA3939',
      clickHandler: goToPreviousPath,
    },
    rightButton: { label: '編集を続ける' },
  });

  return (
    <>
      {/* <header> */}
      <Headroom className='common-header' disableInlineStyles onUnpin={() => setOpen(false)}>
        <div className='flex-container'>
          <div className='flex-item-left'>
            {previousPageName && previousPathname && (
              <a
                onClick={isLifeInpuranceInputPage ? showGoBackAlertDialog : goToPreviousPath}
                role='button'>
                <span className='flex-item-left-arrow'>
                  <StyledArrowLeftSVG color='white' />
                </span>
                <span className='flex-item-left-text'>{previousPageName}</span>
              </a>
            )}
          </div>
          <div className='flex-item-center'>
            <StyledLogoWhiteSVG color='white' width='65.75' height='20' />
          </div>
          <div ref={node} className='flex-item-right'>
            <BurgerComponent open={open} setOpen={setOpen} />
            <MenuComponent open={open} setOpen={setOpen} />
          </div>
        </div>
        <style jsx>
          {`
          .flex-container {
            width: 100%
            position: fixed;
            display: flex;
            width: 100%;
            height: 48px;
            background: #005BAC;
            align-items: center;
          }
          .flex-item-left {
            position: relative;
            width:calc(50% - 65px);
            height: 100%;
            text-align: left;
            overflow: hidden;
          }
          .flex-item-left-arrow {
            position: absolute;
            width: 24px;
            height: 24px;
            top: 12px;
            left: 16px;
          }
          .flex-item-left-text {
            position: absolute;
            width: calc(100% - 48px);
            height: 12px;
            top: 18px;
            left: 48px;
            font-style: normal;
            font-weight: normal;
            font-size: 12px;
            line-height: 100%;
            color: #ffffff;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            margin: 0;
            padding: 0;
          }
          .flex-item-center {
            width: calc(65px + 32px + 32px);
            height: 100%;
            display: flex;
            justify-content: center;
            align-items: center;
          }
          .flex-item-right {
            width: calc(50% - 65px);
            height: 100%;
            text-align: right;
          }
        `}
        </style>
      </Headroom>
    </>
  );
};
