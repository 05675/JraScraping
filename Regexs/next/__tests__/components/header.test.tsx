import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from '@src/components/header';

describe('header', () => {
  test('snapshot', () => {
    const snapshot = renderer
      .create(<Header previousPageName='testPage' previousPathname='http://dummy.xxx/' />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
