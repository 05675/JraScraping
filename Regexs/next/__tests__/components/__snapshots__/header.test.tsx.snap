// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot 1`] = `
<div
  className="common-header headroom-wrapper"
  style={
    Object {
      "height": null,
    }
  }
>
  <div
    className="headroom headroom--unfixed"
  >
    <div
      className="flex-container"
    >
      <div
        className="flex-item-left"
      >
        <a
          onClick={[Function]}
          role="button"
        >
          <span
            className="flex-item-left-arrow"
          >
            <svg>

              className="sc-AxhCb NcLaj"
              color="white"
            />
          </span>
          <span
            className="flex-item-left-text"
          >
            testPage
          </span>
        </a>
      </div>
      <div
        className="flex-item-center"
      >
        <svg>

          className="sc-fzpans dbiFkb"
          color="white"
          height="20"
          width="65.75"
        />
      </div>
      <div
        className="flex-item-right"
      >
        <button
          className="sc-AxirZ eTutho"
          onClick={[Function]}
          open={false}
        >
          <div />
          <div />
          <div />
        </button>
        <nav
          className="sc-AxjAm gLeCXj"
          open={false}
        >
          <a
            href="/"
            onClick={[Function]}
          >
            サインアウト
          </a>
        </nav>
      </div>
    </div>
    <style
      jsx={true}
    >
      
          .flex-container {
            width: 100%
            position: fixed;
            display: flex;
            width: 100%;
            height: 48px;
            background: #005BAC;
            align-items: center;
          }
          .flex-item-left {
            position: relative;
            width:calc(50% - 65px);
            height: 100%;
            text-align: left;
            overflow: hidden;
          }
          .flex-item-left-arrow {
            position: absolute;
            width: 24px;
            height: 24px;
            top: 12px;
            left: 16px;
          }
          .flex-item-left-text {
            position: absolute;
            width: calc(100% - 48px);
            height: 12px;
            top: 18px;
            left: 48px;
            font-style: normal;
            font-weight: normal;
            font-size: 12px;
            line-height: 100%;
            color: #ffffff;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            margin: 0;
            padding: 0;
          }
          .flex-item-center {
            width: calc(65px + 32px + 32px);
            height: 100%;
            display: flex;
            justify-content: center;
            align-items: center;
          }
          .flex-item-right {
            width: calc(50% - 65px);
            height: 100%;
            text-align: right;
          }
        
    </style>
  </div>
</div>
`;
